### DISCUSSION TITLE
Where is the source for the official Python runtime?

### DISCUSSION THREAD

**Discussion by jayhale** [vercel/vercel#6127] [Help] [answered]
Where can I find the source for the official Python runtime / builder for Vercel?

I'd like to continue developing [@ardnt/vercel-python-wsgi](https://github.com/ardnt/vercel-python-wsgi) (previously [@ardnt/now-python-wsgi](https://github.com/ardnt/now-python-wsgi)) and it seems like the right path is to adapt it to being a "runtime" instead of a "builder".  While the existing Python runtime does work, the filesystem-as-an-API approach doesn't work for the projects I work on - the same issue that created the need for [@ardnt/now-python-wsgi](https://github.com/ardnt/now-python-wsgi).

I can't seem to find the Python runtime source in the [Vercel repository](https://github.com/vercel/vercel).  [@vercel/python includes a file `vc_init.py`](https://yarnpkg.com/package/@vercel/python?files) that [is not present](https://github.com/search?q=org%3Avercel+vc_init.py) anywhere in the Vercel repository.  I'm guessing the issue is user error on my part - anyone willing to point me in the right direction?

**Comment by styfle** [comment, marked as answer]
Hi @jayhale 

The source code can be found here: https://github.com/vercel/vercel/tree/main/packages/python

**Reply by jayhale** [reply]
Thanks - fascinating that isn’t indexed for search.  Much appreciated.
